connectors/protocol/connectors.py

```diff
@@ -483,6 +485,11 @@ def document_level_security_enabled(self):
             ["document_level_security", "enabled"], default=False
         )
 
+    def native_connector_api_keys_enabled(self):
```

Comment:
Why is this a feature toggle? Should this be just enabled for all native connectors?

Reply:
> Should this be just enabled for all native connectors?

Yes it should be, I've set it to `False` because these changes are pretty breaking until the other components (ES and Kibana) are ready as well. I intend to change the default to `True`.

> Why is this a feature toggle?

I thought it would be good if we had a way to easily rollback for one release. Perhaps I'm being overly cautious? WDYT?

Reply:
> Yes it should be, I've set it to False because these changes are pretty breaking until the other components (ES and Kibana) are ready as well. I intend to change the default to True.

Yeah, I think it's a good idea. However, we should remove it once all the components are ready. Perhaps the next release?

My main concern is that the customers may get confused to see this feature in their connectors configuration. I think I'm being overly cautious too :). 

Let's keep this as it is 👍🏻